src/finetining/NLLB/finetune.py

```diff
@@ -109,11 +151,13 @@ def get_batch_pairs(batch_size, data=df_train):
         os.mkdir(MODEL_SAVE_PATH)
 
     tq = trange(len(losses), training_steps)
-    def evaluate_model(model, tokenizer, val_dataset, max_length):
+    def evaluate_model(model, tokenizer, val_dataset, max_length, lang1, lang2, l1, l2):
         model.eval()  # Set the model to evaluation mode
         val_losses = []
+        random.shuffle(val_dataset)
         with torch.no_grad():  # Disable gradient calculation
-            for xx, yy, lang1, lang2 in val_dataset:
+            for val_samp in val_dataset:
+                xx, yy = val_samp["translation"][l1], val_samp["translation"][l2]
                 tokenizer.src_lang = lang1
                 x = tokenizer(xx, return_tensors='pt', padding=True, truncation=True, max_length=max_length).to(model.device)
```

Comment:
this x this needs to be passed into the model to and the val_losses is returned.

Reply:
Check it again, the x input was passed to the model and the val_loss was returned

Reply:
Okay then. I was looking at the old commit. But I   also observed that you removed your seed. I didn't say you should you remove it. The huggingface style of setting seed is better than using random.seed.
I'll merge the PR now and handle the seed. Please can you work with Tosin to give you access to the GPU so can run some experiments?
Let's know if the code works fine.